api/src/services/authentication-service.ts

```diff
@@ -3,34 +3,40 @@ import { createNonce, getHexEncodedKey, verifySignedNonce } from '../utils/encry
 import * as AuthDb from '../database/auth';
 import jwt from 'jsonwebtoken';
 import { AuthenticationServiceConfig } from '../models/config/services';
+import { SsiService } from './ssi-service';
+import { User, UserRoles, UserType } from '../models/types/user';
 
 export class AuthenticationService {
-	private readonly userService: UserService;
-	private readonly serverSecret: string;
-	private readonly jwtExpiration: string;
-
-	constructor(userService: UserService, authenticationServiceConfig: AuthenticationServiceConfig) {
-		const { serverSecret, jwtExpiration } = authenticationServiceConfig;
-		this.userService = userService;
-		this.serverSecret = serverSecret;
-		this.jwtExpiration = jwtExpiration;
-	}
+	constructor(
+		private readonly userService: UserService,
+		private readonly ssiService: SsiService,
+		private readonly config: AuthenticationServiceConfig
+	) {}
```

Comment:
Why do you need an empty constructor? 
I noticed this a few times in the PR, but didn't get the point

Reply:
This is something I've found on a different typescript project by adding private readonly to the parameters of the constructor it automatically generates these as class fields, so you can reduce code overhead with this.

Reply:
it is basically the same as writing:

`private readonly userService: UserService;
private readonly serverSecret: string;
private readonly jwtExpiration: string;

constructor(userService: UserService, authenticationServiceConfig: AuthenticationServiceConfig) {
	const { serverSecret, jwtExpiration } = authenticationServiceConfig;
	this.userService = userService;
	this.serverSecret = serverSecret;
	this.jwtExpiration = jwtExpiration;
}`

Reply:
Alright, I didn't know this